File: doc-examples/jdbc-example-java/src/main/java/example/Teacher.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity(value = "TBL_TEACHER", alias = "t")
public class Teacher {
    @Id
    @GeneratedValue(GeneratedValue.Type.IDENTITY)
    private Long id;
}

File: doc-examples/jdbc-example-python/src/main/python/example/Teacher.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity


@MappedEntity(value="TBL_TEACHER", alias="t")
@dataclass
class Teacher:
    id: Annotated[int | None, Id, GeneratedValue("IDENTITY")] = None
